base/abstractset.jl

```diff
@@ -125,9 +127,12 @@ Set{Int64} with 1 element:
   2
 ```
 """
-intersect(s::AbstractSet, itr, itrs...) = intersect!(intersect(s, itr), itrs...)
+function intersect(s::AbstractSet, itr, itrs...)
+    T = promote_eltype(s, itr, itrs...)
+    return intersect!(Set{T}(s), itr, itrs...)
```

Comment:
This is a bug --- it means we give a different container type for 3 arguments than 2 (and a different type than 1.7).
On 1.7:
```
julia> intersect(BitSet([1,2]), [1,2], [2])
BitSet with 1 element:
  2

julia> intersect(BitSet([1,2]), [1,2])
BitSet with 2 elements:
  1
  2
```
on master:
```
julia> intersect(BitSet([1,2]), [1,2], [2])
Set{Int64} with 1 element:
  2

julia> intersect(BitSet([1,2]), [1,2])
BitSet with 2 elements:
  1
  2
```